src/libs/actions/Transaction.ts

```diff
@@ -128,8 +128,17 @@ function removeWaypoint(transaction: Transaction, currentIndex: string, isDraft:
     // Doing a deep clone of the transaction to avoid mutating the original object and running into a cache issue when using Onyx.set
     let newTransaction: Transaction = {
         ...transaction,
+        amount: transaction?.amount ?? 0,
+        billable: transaction?.billable ?? false,
+        category: transaction?.category ?? '',
+        created: transaction?.created ?? '',
+        currency: transaction?.currency ?? '',
+        merchant: transaction?.merchant ?? '',
+        reportID: transaction?.reportID ?? '',
+        transactionID: transaction?.transactionID ?? '',
+        tag: transaction?.tag ?? '',
```

Comment:
Maybe this way it will be more safe
```suggestion
    let newTransaction: Transaction = {
        // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
        ...transaction as Transaction,
```

Reply:
yeah good idea!